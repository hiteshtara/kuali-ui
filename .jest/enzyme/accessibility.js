/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
