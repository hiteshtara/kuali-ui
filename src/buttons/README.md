# Buttons

Set of button components to use in Kuali's UI.

## `<RaisedButton />`

### PropTypes
| name      |type       | default     | required      |
|-----------|-----------|-------------|---------------|
| variant   | string    | default     | yes           |

### Example Usage

```js
import { RaisedButton, Icon } from 'kuali-ui'

export function SomeComponent() {
  return (
    <RaisedButton variant='primary' label='Click Me' />
  )
}

```

## `<FlatButton />`

### PropTypes
| name      |type       | default     | required      |
|-----------|-----------|-------------|---------------|
| variant   | string    | default     | yes           |

### Example Usage

```js
import { FlatButton } from 'kuali-ui'

export function SomeComponent() {
  return (
    <FlatButton variant='primary' label='Click Me' />
  )
}

```

## `<IconButton />`

### PropTypes
| name      |type       | default     | required      |
|-----------|-----------|-------------|---------------|
| variant   | string    | default     | yes           |

### Example Usage

```js
import { IconButton } from 'kuali-ui'

export function SomeComponent() {
  return (
    <IconButton variant='primary'>close</IconButton>
  )
}

```

## `<ButtonGroup />`

### PropTypes
| name      |type       | default     | required      |
|-----------|-----------|-------------|---------------|
| children  | array     | none        | yes           |

### Example Usage

```js
import { ButtonGroup, RaisedButton } from 'kuali-ui'

export function SomeComponent() {
  return (
    <ButtonGroup>
      <RaisedButton label='First' />
      <RaisedButton label='Second' />
      <RaisedButton label='Third' />
    </ButtonGroup>
  )
}

```
