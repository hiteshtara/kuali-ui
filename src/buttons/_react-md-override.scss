/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

// NOTE: other components use reactmd buttons internally;
// leave this file here until we figure out what to do

$md-btn-include-floating: false;
$md-btn-lr-padding: 10px;
$md-btn-min-width: 0;

@include react-md-buttons;
