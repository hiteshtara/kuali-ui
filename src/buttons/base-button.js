/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { injectInk } from 'react-md'
import classnames from 'classnames'

import { unsupported as UnsupportedProp } from '../utils/proptypes'
import Icon from '../icons'
import injectTooltip from '../tooltips'

export const supportedVariants = [
  'default',

  // all purpose
  'clear',
  'plain',

  // brand colors
  'primary',

  // actions
  'error',
  'info',
  'success',
  'warning',

  // brand colors
  'wintergreen',
  'frost',
  'glacier',
  'mint',
  'popsicle',
  'magma',
  'chili',
  'flare',
  'torch'
]

const unsupportedProps = [
  'forceIconSize',
  'forceIconFontSize',
  'floating',
  'fixed',
  'fixedPosition',
  'mini',
  'primary',
  'secondary'
].reduce((acc, cur) => ({ ...acc, [cur]: UnsupportedProp }), {})

@injectInk
@injectTooltip
export default class Button extends PureComponent {
  static defaultProps = {
    block: false,
    children: null,
    component: 'button',
    icon: false,
    iconBefore: true,
    size: 'default',
    type: 'submit',
    variant: 'default'
  }

  static propTypes = {
    ...unsupportedProps,
    block: PropTypes.bool.isRequired,
    children: PropTypes.node,
    className: PropTypes.string,
    component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
    flat: PropTypes.bool,
    icon: PropTypes.bool,
    iconBefore: PropTypes.bool.isRequired,
    iconClassName: PropTypes.string,
    ink: PropTypes.node,
    innerRef: PropTypes.func,
    label: PropTypes.node,
    raised: PropTypes.bool,
    size: PropTypes.oneOf(['default', 'small']).isRequired,
    type: PropTypes.oneOf(['button', 'reset', 'submit']),
    variant: PropTypes.oneOf(supportedVariants).isRequired
  }

  get icon () {
    let { children: stringOrIcon, iconClassName } = this.props

    if (!stringOrIcon) return null

    const isString = typeof stringOrIcon === 'string'
    const ToRender = isString ? <Icon name={stringOrIcon} /> : stringOrIcon

    return React.cloneElement(ToRender, {
      className: classnames(
        'kuali-button__icon',
        ToRender.props.className,
        iconClassName
      )
    })
  }

  render () {
    const {
      block,
      className,
      component: Component,
      flat,
      label,
      icon,
      iconBefore,
      ink,
      innerRef,
      raised,
      size,
      tooltip,
      variant,
      ...rest
    } = this.props

    delete rest.children
    delete rest.iconClassName
    delete rest.tooltipLabel
    delete rest.tooltipDelay
    delete rest.tooltipPosition

    const classes = classnames(
      'kuali-button',
      `kuali-button--${variant}`,
      `kuali-button--size__${size}`,
      {
        'kuali-button--block': block,
        'kuali-button--flat': flat,
        'kuali-button--icon': icon,
        'kuali-button--raised': raised
      },
      className
    )

    if (innerRef) {
      rest.ref = innerRef
    }

    return (
      <Component className={classes} {...rest}>
        {ink}
        {tooltip}

        {iconBefore && this.icon}
        {label && <span className='kuali-button__label'>{label}</span>}
        {!iconBefore && this.icon}
      </Component>
    )
  }
}
