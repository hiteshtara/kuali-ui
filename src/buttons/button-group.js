/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class ButtonGroup extends PureComponent {
  static propTypes = {
    children: PropTypes.arrayOf(PropTypes.node).isRequired
  }

  render () {
    return (
      <div className='kuali-button-group'>
        {this.props.children}
      </div>
    )
  }
}
