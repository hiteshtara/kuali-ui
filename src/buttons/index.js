/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import Button from './base-button'

export default Button
export { default as ButtonGroup } from './button-group'
export { default as RaisedButton } from './raised-button'
export { default as FlatButton } from './flat-button'
export { default as IconButton } from './icon-button'
