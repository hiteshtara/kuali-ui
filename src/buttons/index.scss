/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

@import "./react-md-override";

$kuali-flat-button-text-color: #555;

$kuali-raised-button-shadow: 0 2px 0 rgba(0, 0, 0, 0.3);

$kuali-button-color: #ECECEC;
$kuali-button-text-color: #555;
$kuali-button-hover-color: darken($kuali-button-color, 5);
$kuali-button-pressed-color: darken($kuali-button-color, 10);

.kuali-button {
  position: relative;
  background-color: $kuali-button-color;
  color: $kuali-button-text-color;
  border: 0;
  border-radius: 2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 8px 15px 8px 15px;
  font-weight: 500;
  vertical-align: middle;
  text-decoration: none;
  cursor: pointer;

  * {
    pointer-events: none;
  }

  & > &__icon {
    margin-top: -2px;
    margin-bottom: -1px;

    @include media(">=desktop") {
      margin-top: -3px;
      margin-bottom: -1px;
    }
  }

  & > &__label {
    font-size: 14px;
  }

  & + & {
    margin-left: 5px;
  }

  &:focus {
    outline: none;
  }

  &:hover {
    background-color: $kuali-button-hover-color;
  }

  &:active {
    background-color: $kuali-button-pressed-color;
  }

  &__label + &__icon,
  &__icon + &__label {
    margin-left: 8px;
  }

  .kuali-button-group & {
    margin-left: 0;
    margin-right: 0;
    border-radius: 0;
  }

  &:disabled {
    cursor: not-allowed;
    background-color: $kuali-button-color;
    color: lighten($kuali-button-text-color, 20);

    .kuali-button__icon {
      color: inherit;
    }

    &:active {
      background-color: $kuali-button-color;
    }

    &:focus {
      outline: none;
    }

    &:hover {
      background-color: $kuali-button-color;
    }
  }

  .kuali-button-group &:first-of-type {
    border-bottom-left-radius: 2px;
    border-top-left-radius: 2px;
  }

  .kuali-button-group &:last-of-type {
    border-bottom-right-radius: 2px;
    border-top-right-radius: 2px;
  }
}

// ===========
// Block(full-width)
// ===========
.kuali-button--block {
  display: flex;
  width: 100%;
  margin-bottom: 20px;
}

.kuali-button + .kuali-button--block {
  margin-left: 0;
}

// ===========
// Sizes
// ===========
.kuali-button--size__small {
  @include media(">=desktop") {
    padding: 4px 10px 3px 10px;

    .kuali-button__icon {
      font-size: 16px;
      margin-top: 0;
      margin-bottom: 0;
    }

    .kuali-button__label {
      font-size: 12px;
    }
  }
}

// ===========
// Types
// ===========
.kuali-button--raised {
  position: relative;
  box-shadow: inset 0 -2px 0 rgba(0, 0, 0, 0.25);
  transition: top ease-in-out 0.3s;

  &:active {
    box-shadow: none;
    top: 1px;
  }
}

.kuali-button--icon {
  background-color: transparent;
  border-radius: 50%;
  padding: 14px 12px 12px;

  &:hover {
    background-color: $kuali-button-hover-color;
  }

  &:active {
    background-color: $kuali-button-pressed-color;
  }
}

// ===========
// Variant Builder
// ===========
@mixin buttonVariant($name, $paletteName: null) {
  $paletteName: $name !default;
  $variant-background: palette($paletteName);
  $variant-text-color: #FFF;

  @if lightness($variant-background) > 80 {
    $variant-text-color: $kuali-button-text-color;
  }

  .kuali-button--#{$name} {
    color: $variant-text-color;
    background-color: $variant-background;

    &:hover {
      background-color: darken($variant-background, 5);
    }

    &:active {
      background-color: darken($variant-background, 10);
    }

    .kuali-button__icon {
      color: inherit;
    }

    &.kuali-button--icon {
      color: $variant-background;
      background-color: transparent;

      &:hover {
        background-color: lighten($variant-background, 30);
      }

      &:active {
        background-color: lighten($variant-background, 20);
      }
    }
  }
}

// ===========
// Variants
// ===========
.kuali-button--clear {
  background-color: transparent;

  &:hover {
    background-color: transparent;
  }

  &:active {
    background-color: transparent;
  }
}

.kuali-button--plain {
  background-color: #FFF;
}

@include buttonVariant("error", "chili");
@include buttonVariant("info", "glacier");
@include buttonVariant("success", "mint");
@include buttonVariant("warning", "torch");
// NOTE: glacier isn't technically the primary color but it is that way for buttons
@include buttonVariant("primary", "glacier");
@include buttonVariant("wintergreen");
@include buttonVariant("frost");
@include buttonVariant("glacier");
@include buttonVariant("mint");
@include buttonVariant("popsicle");
@include buttonVariant("magma");
@include buttonVariant("chili");
@include buttonVariant("flare");
@include buttonVariant("chili");
@include buttonVariant("torch");
