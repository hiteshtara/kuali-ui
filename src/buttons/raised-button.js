/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import BaseButton from './base-button'

export default function RaisedButton (props) {
  return <BaseButton raised {...props} />
}
