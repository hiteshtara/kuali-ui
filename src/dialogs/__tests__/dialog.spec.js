/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import { RaisedButton } from '../../buttons'
import Dialog from '..'

describe('Dialog Component', () => {
  it('should render with a close button', () => {
    const wrapper = shallow(
      <Dialog
        id='modal-dialog'
        visible
        onHide={() => {}}
        title='Modal Dialog'
        showClose
      >
        <div>Some really nice modal content goes here!</div>
      </Dialog>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render without a close button', () => {
    const wrapper = shallow(
      <Dialog id='modal-dialog' visible onHide={() => {}} title='Modal Dialog'>
        <div>Some really nice modal content goes here!</div>
      </Dialog>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render without header', () => {
    const wrapper = shallow(
      <Dialog
        id='modal-dialog'
        visible
        onHide={() => {}}
        title='Modal Dialog'
        showHeader={false}
      >
        <div>Some really nice modal content goes here!</div>
      </Dialog>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render without footer', () => {
    const wrapper = shallow(
      <Dialog
        id='modal-dialog'
        visible
        onHide={() => {}}
        title='Modal Dialog'
        showFooter={false}
      >
        <div>Some really nice modal content goes here!</div>
      </Dialog>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render without header or footer', () => {
    const wrapper = shallow(
      <Dialog
        id='modal-dialog'
        visible
        onHide={() => {}}
        title='Modal Dialog'
        showHeader={false}
        showFooter={false}
      >
        <div>Some really nice modal content goes here!</div>
      </Dialog>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with actions positioned right', () => {
    const actions = [
      <RaisedButton label='first' />,
      <RaisedButton label='second' />
    ]
    const wrapper = shallow(
      <Dialog
        id='modal-dialog'
        visible
        onHide={() => {}}
        title='Modal Dialog'
        actionsPosition='right'
        actions={actions}
      >
        <div>Some really nice modal content goes here!</div>
      </Dialog>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with actions positioned left', () => {
    const actions = [
      <RaisedButton label='first' />,
      <RaisedButton label='second' />
    ]
    const wrapper = shallow(
      <Dialog
        id='modal-dialog'
        visible
        onHide={() => {}}
        title='Modal Dialog'
        actionsPosition='left'
        actions={actions}
      >
        <div>Some really nice modal content goes here!</div>
      </Dialog>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
