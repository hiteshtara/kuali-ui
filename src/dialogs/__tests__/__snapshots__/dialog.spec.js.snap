// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component should render with a close button 1`] = `
<DialogContainer
  aria-describedby="Modal Dialog"
  autopadContent={true}
  autosizeContent={true}
  closeOnEsc={true}
  component={[Function]}
  contentClassName="kuali-dialog-section"
  contentComponent="section"
  defaultVisibleTransitionable={false}
  dialogClassName="kuali-dialog kuali-dialog--actions-center"
  focusOnMount={false}
  id="modal-dialog"
  lastChild={true}
  onHide={[Function]}
  transitionEnterTimeout={300}
  transitionLeaveTimeout={300}
  visible={true}
>
  <header
    className="kuali-dialog--header"
  >
    <h1
      className="kuali-dialog--header---title"
    >
      Modal Dialog
    </h1>
    <div
      className="kuali-dialog--header---close"
    >
      <IconButton
        onClick={[Function]}
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="close"
          variant="default"
        />
      </IconButton>
    </div>
  </header>
  <div
    className="kuali-dialog--content"
  >
    <div>
      Some really nice modal content goes here!
    </div>
  </div>
</DialogContainer>
`;

exports[`Dialog Component should render with actions positioned left 1`] = `
<DialogContainer
  actions={
    Array [
      <RaisedButton
        label="first"
      />,
      <RaisedButton
        label="second"
      />,
    ]
  }
  aria-describedby="Modal Dialog"
  autopadContent={true}
  autosizeContent={true}
  closeOnEsc={true}
  component={[Function]}
  contentClassName="kuali-dialog-section"
  contentComponent="section"
  defaultVisibleTransitionable={false}
  dialogClassName="kuali-dialog kuali-dialog--actions-left kuali-dialog--actions-center"
  focusOnMount={false}
  id="modal-dialog"
  lastChild={true}
  onHide={[Function]}
  transitionEnterTimeout={300}
  transitionLeaveTimeout={300}
  visible={true}
>
  <header
    className="kuali-dialog--header"
  >
    <h1
      className="kuali-dialog--header---title"
    >
      Modal Dialog
    </h1>
    <div
      className="kuali-dialog--header---close"
    >
      <IconButton
        onClick={[Function]}
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="close"
          variant="default"
        />
      </IconButton>
    </div>
  </header>
  <div
    className="kuali-dialog--content"
  >
    <div>
      Some really nice modal content goes here!
    </div>
  </div>
</DialogContainer>
`;

exports[`Dialog Component should render with actions positioned right 1`] = `
<DialogContainer
  actions={
    Array [
      <RaisedButton
        label="first"
      />,
      <RaisedButton
        label="second"
      />,
    ]
  }
  aria-describedby="Modal Dialog"
  autopadContent={true}
  autosizeContent={true}
  closeOnEsc={true}
  component={[Function]}
  contentClassName="kuali-dialog-section"
  contentComponent="section"
  defaultVisibleTransitionable={false}
  dialogClassName="kuali-dialog kuali-dialog--actions-right kuali-dialog--actions-center"
  focusOnMount={false}
  id="modal-dialog"
  lastChild={true}
  onHide={[Function]}
  transitionEnterTimeout={300}
  transitionLeaveTimeout={300}
  visible={true}
>
  <header
    className="kuali-dialog--header"
  >
    <h1
      className="kuali-dialog--header---title"
    >
      Modal Dialog
    </h1>
    <div
      className="kuali-dialog--header---close"
    >
      <IconButton
        onClick={[Function]}
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="close"
          variant="default"
        />
      </IconButton>
    </div>
  </header>
  <div
    className="kuali-dialog--content"
  >
    <div>
      Some really nice modal content goes here!
    </div>
  </div>
</DialogContainer>
`;

exports[`Dialog Component should render without a close button 1`] = `
<DialogContainer
  aria-describedby="Modal Dialog"
  autopadContent={true}
  autosizeContent={true}
  closeOnEsc={true}
  component={[Function]}
  contentClassName="kuali-dialog-section"
  contentComponent="section"
  defaultVisibleTransitionable={false}
  dialogClassName="kuali-dialog kuali-dialog--actions-center"
  focusOnMount={false}
  id="modal-dialog"
  lastChild={true}
  onHide={[Function]}
  transitionEnterTimeout={300}
  transitionLeaveTimeout={300}
  visible={true}
>
  <header
    className="kuali-dialog--header"
  >
    <h1
      className="kuali-dialog--header---title"
    >
      Modal Dialog
    </h1>
    <div
      className="kuali-dialog--header---close"
    >
      <IconButton
        onClick={[Function]}
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="close"
          variant="default"
        />
      </IconButton>
    </div>
  </header>
  <div
    className="kuali-dialog--content"
  >
    <div>
      Some really nice modal content goes here!
    </div>
  </div>
</DialogContainer>
`;

exports[`Dialog Component should render without footer 1`] = `
<DialogContainer
  actions={null}
  aria-describedby="Modal Dialog"
  autopadContent={true}
  autosizeContent={true}
  closeOnEsc={true}
  component={[Function]}
  contentClassName="kuali-dialog-section"
  contentComponent="section"
  defaultVisibleTransitionable={false}
  dialogClassName="kuali-dialog kuali-dialog--actions-center"
  focusOnMount={false}
  id="modal-dialog"
  lastChild={true}
  onHide={[Function]}
  transitionEnterTimeout={300}
  transitionLeaveTimeout={300}
  visible={true}
>
  <header
    className="kuali-dialog--header"
  >
    <h1
      className="kuali-dialog--header---title"
    >
      Modal Dialog
    </h1>
    <div
      className="kuali-dialog--header---close"
    >
      <IconButton
        onClick={[Function]}
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="close"
          variant="default"
        />
      </IconButton>
    </div>
  </header>
  <div
    className="kuali-dialog--content"
  >
    <div>
      Some really nice modal content goes here!
    </div>
  </div>
</DialogContainer>
`;

exports[`Dialog Component should render without header 1`] = `
<DialogContainer
  aria-describedby="Modal Dialog"
  autopadContent={true}
  autosizeContent={true}
  closeOnEsc={true}
  component={[Function]}
  contentClassName="kuali-dialog-section"
  contentComponent="section"
  defaultVisibleTransitionable={false}
  dialogClassName="kuali-dialog kuali-dialog--actions-center"
  focusOnMount={false}
  id="modal-dialog"
  lastChild={true}
  onHide={[Function]}
  transitionEnterTimeout={300}
  transitionLeaveTimeout={300}
  visible={true}
>
  <div
    className="kuali-dialog--content"
  >
    <div>
      Some really nice modal content goes here!
    </div>
  </div>
</DialogContainer>
`;

exports[`Dialog Component should render without header or footer 1`] = `
<DialogContainer
  actions={null}
  aria-describedby="Modal Dialog"
  autopadContent={true}
  autosizeContent={true}
  closeOnEsc={true}
  component={[Function]}
  contentClassName="kuali-dialog-section"
  contentComponent="section"
  defaultVisibleTransitionable={false}
  dialogClassName="kuali-dialog kuali-dialog--actions-center"
  focusOnMount={false}
  id="modal-dialog"
  lastChild={true}
  onHide={[Function]}
  transitionEnterTimeout={300}
  transitionLeaveTimeout={300}
  visible={true}
>
  <div
    className="kuali-dialog--content"
  >
    <div>
      Some really nice modal content goes here!
    </div>
  </div>
</DialogContainer>
`;
